docs/examples/ex27.py

```diff
@@ -0,0 +1,237 @@
+from skfem import *
+from skfem.models.poisson import vector_laplace, laplace
+from skfem.models.general import divergence, rot
+
+from functools import partial
+from itertools import cycle, islice
+from typing import Tuple, Iterable
+
+from matplotlib.pyplot import subplots
+from matplotlib.tri import Triangulation
+import numpy as np
+from scipy.sparse import bmat, block_diag, csr_matrix
+
+from pygmsh import generate_mesh
+from pygmsh.built_in import Geometry
+
+from pacopy import natural
+
+
+@linear_form
+def acceleration(v, dv, w):
+    """Compute the vector (v, u . grad u) for given velocity u."""
+    u, du = w.w, w.dw
+    # TODO: Handle the indices more cleverly
```

Comment:
Resolve TODO before merge.

Reply:
Yes.

I'm struggling a bit with this.  I'd like to write this so that it was independent of the number of spatial dimensions instead of only working for two, as here.  And then probably move the function to `skfem.models.general` or a new hydrodynamical submodule of `skfem.models`.

I've made a slight improvement with
```python
    # return (v[0] * (u[0] * du[0][0] + u[1] * du[0][1])
    #         + v[1] * (u[0] * du[1][0] + u[1] * du[1][1]))
    return sum(u * du[0]) * v[0] + sum(u * du[1]) * v[1]
 ```
but that's still two-dimensional.  Perhaps I should be using [`numpy.einsum`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.einsum.html) here.

Reply:
This is a bit [above](https://github.com/kinnala/scikit-fem/pull/145#issuecomment-475484311).  This time, I should work out properly all the indices on u, dv, w.w, and w.dw, how they relate to components of velocity, components of gradient, and quadrature point, and what shape is required in the output, and document all that.

Reply:
I'm just guessing here:
```python
return sum(np.einsum('ij...,j...', du, u), v)
```

Reply:
The reason why you need the ellipsis here is that the last dimension corresponds to the number of quadrature points and the second-to-last dimension is the number of elements. The shape of `u` is `Ndim x Nelems x Nqp` and the shape of `du` is `Ndim x Ndim x Nelems x Nqp`.

Reply:
Let me add that the output shape should be `Nelems x Nqp`.

Reply:
Thank you.  That looks good.  I'll study einsum more closely so that I actually know what I'm doing.  I see that it has two different modes, ‘explicit’ and ‘implicit’, depending on whether the indices of the result are specified; ‘explicit’, as in your [suggestion](https://github.com/kinnala/scikit-fem/pull/145#discussion_r279053772) looks clearer.  

I wonder whether it would also be able to subsume the `sum` with `v`, and then whether that would be more or less clear.  It might be more explicit, instead of having to remember or work out which index the `sum` was summing over.  I'll experiment.

Reply:
Oh, actually it's even easier than that since the elements and abscissæ are the trailing two indices for all three entities; thus, a trailing ellipsis for all three and no need to explicitly state the indices of the output:
```python
np.einsum('j...,ij...,i...', u, du, v)
```

Reply:
That seems reasonably close to the Cartesian tensorial notation ignoring the discretization: $u_j u_{i,j} v_i$.

Reply:
A single `einsum` can't be used for the bilinear form though, so it's either 
```python
    return sum((np.einsum('j...,ij...->i...', w.w, du)
                     + np.einsum('j...,ij...->i...', u, w.dw)) * v)
```
or
```python
np.einsum('j...,ij...,i...', w.w, du, v) + np.einsum('j...,ij...,i...', u, w.dw, v)
```
Not much between them, so I've gone with your `sum`–`einsum` suggestion for both linear and bilinear.  Thanks again.  

I think this is good to go now.  

I'll leave these forms in this example for now, moving them to `skfem.models` later if they're needed again in future examples.